app/src/main/java/org/covidwatch/android/ui/MainActivity.kt

```diff
@@ -118,33 +121,41 @@ class MainActivity : AppCompatActivity() {
      *
      * TODO add GPS initialization here, for now we just ask for location permissions
      */
-    @RequiresApi(api = Build.VERSION_CODES.M)
-    public fun initLocationManager() {
-        val permissionCheck = ContextCompat.checkSelfPermission(
-            this,
-            Manifest.permission.ACCESS_FINE_LOCATION
-        )
-        if (permissionCheck != PackageManager.PERMISSION_GRANTED) {
-            if (ActivityCompat.shouldShowRequestPermissionRationale(
-                    this,
-                    Manifest.permission.ACCESS_FINE_LOCATION
-                )
-            ) {
-                Toast.makeText(
-                    this,
-                    getString(R.string.ble_location_permission),
-                    Toast.LENGTH_SHORT
-                ).show()
-            } else {
-                requestPermissions(
-                    arrayOf(
-                        Manifest.permission.ACCESS_COARSE_LOCATION,
-                        Manifest.permission.ACCESS_FINE_LOCATION
-                    ), 1
-                )
+    private val LOCATION_REQUEST_CODE = 1
+    fun initLocationManager() {
+        runOnUiThread {
+            val permissionStatus = ContextCompat.checkSelfPermission(
+                this,
+                ACCESS_FINE_LOCATION
+            )
+            when (permissionStatus){
+                PackageManager.PERMISSION_GRANTED -> vm.permissionRequestResultLiveData.value = true
+                PackageManager.PERMISSION_DENIED -> {
```

Comment:
Nice, so what happens in the UX? Do they go to the next screen?

Reply:
They go to the next screen as soon as they grant permission in the system dialogue.